src/main/python/deep_qa/layers/entailment_models/tuple_match.py

```diff
@@ -0,0 +1,180 @@
+        super(TupleMatch, self).__init__(**params)
+
+    def score_tuple_entailment(self, combined_tuple_input):
+        """
+        Here we take the combined tuple-based entailment input and decide whether it is true or false. The
+        combined input has shape (batch_size, combined_input_dim).  We do not know what
+        combined_input_dim is, as it depends on the combiner.
+        """
+        hidden_input = combined_tuple_input
+        for layer in self.hidden_layers:
+            hidden_input = layer(hidden_input)
+        score_output = self.score_layer(hidden_input)
+        return score_output
+
+
+    def get_output_shape_for(self, input_shapes):
+        #pylint: disable=unused-argument
+        return (1,)
+
+
+    def build(self, input_shape):
+        # Build the dense layer(s)
+        self.hidden_layers = []
+        dense_input_shape = (None, self.num_entailment_features)
+        for i in range(self.num_hidden_layers):
+            curr_dense = Dense(output_dim=self.hidden_layer_width,
+                               activation=self.hidden_layer_activation,
+                               bias=self.use_biases,
+                               name='tuple_match_hidden_layer_%d' % i)
+            curr_dense.build(input_shape=dense_input_shape)
+            dense_input_shape = (None, self.hidden_layer_width)
+            self.hidden_layers.append(curr_dense)
+        score_layer_dense = Dense(output_dim=1,
+                                  activation='sigmoid',
+                                  name='tuple_match_score',
+                                  bias=self.use_biases)
+        score_layer_dense.build(input_shape=dense_input_shape)
+        self.score_layer = score_layer_dense
+
+        # Add the weights from these layers to TupleMatch's trainable weights
+        for hidden in self.hidden_layers + [self.score_layer]:
+            self.trainable_weights.extend(hidden.trainable_weights)
+
+        super(TupleMatch, self).build(input_shape)
+
+    @overrides
+    def compute_mask(self, inputs, input_mask=None):  # pylint: disable=unused-argument
+        # Here, input_mask is optionally a list of two masks (corresponding to the list of inputs) or None.
+        # Each of the masks in input_mask has shape: (batch size, num_slots)
+        # We want to see if either of the inputs is all padding (i.e. the mask would be all 0s), if so, then the
+        # whole tuple_match should be masked, so we would return a 0, otherwise we return a 1.  As such, the
+        # shape of the returned mask is (batch size, 1)
+        if any(input_mask):
+            mask1 = input_mask[0]
+            mask2 = input_mask[1]
+            mask1 = K.cast(K.any(mask1), 'int32')
+            mask2 = K.cast(K.any(mask2), 'int32')
+            return K.cast((mask1 + mask2) >= 2, 'int32')
+
+        return None
+
+    def call(self, inputs, mask=None):
+        tuple1_input = inputs[0]    # shape: (batch size, num_slots, num_slot_words_t1)
```

Comment:
Here also, you can just do `tuple1_input, tuple2_input = inputs`.  Except you should probably use `x` instead of `inputs`, even though x isn't a great name, to match the keras API.  There are some situations in which it actually matters that the names of the arguments match the super class method.